### PR TITLE
Lecture 2

### PR DESCRIPTION
**Pull request by SrBIG**

### CODE REVIEW

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java [hunk 1]
```diff
@@ -23,22 +38,32 @@ public Product getProduct(Long id) throws ProductNotFoundException, IllegalArgum
...
+    synchronized public List<Product> findProducts(String query, String sortBy, String order) {
+        List<Product> actualProducts;
+
+        if (query == null || query.trim().length() == 0) {
+            actualProducts = findAllProducts();
+        } else {
+            actualProducts = findProductsByQuery(query);
+        }
+
+        if (sortBy == null || order == null) {
```
Comment (nikifarau) [inline src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: order may have a default value

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java [hunk 2]
```diff
@@ -54,4 +79,55 @@ private void checkId(Long id) throws IllegalArgumentException {
...
+
+    private List<Product> findAllProducts() {
+        return products.stream()
+                .filter(isValidProduct())
+                .collect(Collectors.toList());
+    }
+
+    private List<Product> findProductsByQuery(String query) {
+        Map<Product, Integer> productsByQuery = new HashMap<>();
+        String[] queries = query.split("\\s");
```
Comment (nikifarau) [inline src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: \\\\s+

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java (hunk 1 above)
Comment (SrBIG) [reply src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: If order isn't important, parameter from get request will be null

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java (hunk 2 above)
Comment (SrBIG) [reply src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: Thanks for advice. I have corrected.

**Pull request merged**